hs_core/views/__init__.py

```diff
@@ -276,14 +276,7 @@ def add_metadata_element(request, shortkey, element_name, *args, **kwargs):
         # seems the user wants to delete all keywords - no need for pre-check in signal handler
         res.metadata.subjects.all().delete()
         is_add_success = True
-        if not res.can_be_public_or_discoverable:
-            res.raccess.public = False
-            res.raccess.discoverable = False
-            res.raccess.save()
-        elif not res.raccess.discoverable:
-            res.raccess.discoverable = True
-            res.raccess.save()
-
+        res.update_public_and_discoverable()
```

Comment:
See if you want to call this also here or leave a TODO comment here: https://github.com/hydroshare/hydroshare/blob/d49c4ceec9e66489daa5185b659bbddf1c62e503/hs_core/views/__init__.py#L470